public/js/src/module/records-queue.js

```diff
@@ -288,38 +287,31 @@ function uploadQueue(byUser = false) {
                                 errorMsg =
                                     result.message ||
                                     gui.getErrorResponseMsg(result.status);
-                                fails.push(record.name);
                                 uploadProgress.update(
                                     record.instanceId,
                                     'error',
                                     errorMsg
                                 );
                             })
-                            .then(() => {
-                                $uploadButton.btnBusyState(false);
-                                if (
-                                    successes.length + fails.length ===
-                                    records.length
-                                ) {
```

Comment:
I do not know what I was thinking when I wrote this code long ago. I'm hoping that way back this made sense, but think it's better I don't find out.

This whole block should just move to after all submissions are completed/failed.